index.js

```diff
@@ -1,68 +1,47 @@
 /* jshint node: true, -W030 */
 
 /*
-	MIT License http://www.opensource.org/licenses/mit-license.php
-	Author Kiril Vatev @catdad
-*/
+    MIT License http://www.opensource.org/licenses/mit-license.php
+    Author Kiril Vatev @catdad
+ */
 
-var path = require("path");
-var loaderUtils = require("loader-utils");
-var SourceListMap = require("source-list-map").SourceListMap;
-var CleanCSS = require("clean-css");
+var path = require('path');
+var loaderUtils = require('loader-utils');
+var CleanCSS = require('clean-css');
+var RequestShortener = require('webpack/lib/RequestShortener');
+
+module.exports = function (content, map) {
+    var config = loaderUtils.getLoaderConfig(this, 'cssRawLoader');
 
-module.exports = function(content, map) {
-	var query = loaderUtils.parseQuery(this.query);
-    
     this.cacheable && this.cacheable();
-	this.value = content;
-    
-    if (typeof map !== 'string') {
-        map = JSON.stringify(map);
-    }
-    
-    var result = [];
-    
-    var minimize = (this && this.minimize) || false;
-    
-    if(minimize) {
-		var options = Object.create(query);
-		if(query.sourceMap && map) {
-			options.sourceMap = map;
-		}
-		var minimizeResult = new CleanCSS(options).minify(content);
-		map = minimizeResult.sourceMap;
-		content = minimizeResult.styles;
-		if(typeof map !== "string") {
-			map = JSON.stringify(map);
+    this.value = content;
+
+    if (config.minimize || this.minimize && config.minimize !== false) {
+        var options = Object.create(config);
+        if (config.sourceMap || this.sourceMap && config.sourceMap !== false) {
+            options.sourceMap = typeof map === 'string' ? map : (map ? JSON.stringify(map) : true);
+        }
+        var minimizeResult = new CleanCSS(options).minify(content);
+        map = JSON.stringify(minimizeResult.sourceMap); // Make the SourceMapGenerator a plain object
+        if (map) {
+            map = JSON.parse(map);
         }
-	} else {
-        var cssRequest = loaderUtils.getRemainingRequest(this);
-        var request = loaderUtils.getCurrentRequest(this);
+        content = minimizeResult.styles;
+    }
 
-        var sourceMap = new SourceListMap();
-        sourceMap.add(content, cssRequest, content);
-        map = sourceMap.toStringWithSourceMap({
-            file: request
-        }).map;
+    if (map && map.sources) {
+        var context = config.context || this.options.context;
+        var requestShortener = new RequestShortener(context);
+        map.sources = map.sources.map(function (source) {
+            return requestShortener.shorten(path.resolve(context, source));
```

Comment:
This is accomplishing the same thing as before, just a bit more succinctly since it uses webpack's `RequestShortener`. I'm providing webpack with a consistent absolute path to a resource so we get a tidy relative path (e.g. `node_modules/angular-material/angular-material.scss` becomes `~/angular-material/angular-material.scss`).